CMakeLists.txt

```diff
@@ -109,6 +109,8 @@ option(ENABLE_CXX_DEPS "Extra library dependencies in srt.pc for the CXX librari
 option(USE_STATIC_LIBSTDCXX "Should use static rather than shared libstdc++" OFF)
 option(ENABLE_INET_PTON "Set to OFF to prevent usage of inet_pton when building against modern SDKs while still requiring compatibility with older Windows versions, such as Windows XP, Windows Server 2003 etc." ON)
 option(USE_OPENSSL_PC "Use pkg-config to find OpenSSL libraries" ON)
+option(USE_BUSY_WAITING "Enable more accurate sending times at a cost of potentially higher CPU load" OFF)
+option(ENABLE_UNEVEN_PACING "Enable and don't compensate uneven pacing of the sending (deviations of inter sending times)" OFF)
```

Comment:
Isn't "odd" the opposite word for "even"?

Although I think the word that could better represent "deviations" would rather be "varying".

Reply:
- `ENABLE_IRREGULAR_PACING`
- `ENABLE_VARYING_PACING`
- `ENABLE_PACING_DEVIATION`

I still like `ENABLE_PACING_DEVIATION` most.

Reply:
- `ENABLE_PACING_DIVERGENCE`

Reply:
Well, "deviation" is still better than "divergence".

But inversion of the boolean meaning might help even more - options must begin with ENABLE, but can be set to true by default. So, to "enable this feature" you might want to say `--disable-send-period-compensation` (so the option would be `ENABLE_SEND_PERIOD_COMPENSATION`.

Reply:
Previously `NO_BUSY_WAITING` was defined in srt.h and udt.h, building SRT core without BUSY waiting and with "send period compensation".
I did this inversion to be preserve the default behavior without specifying those options. So if no options are provided, the core will be built without busy waiting and **with** send period compensation.
If `SEND_PERIOD_COMPENSATION` is introduced, it has to be ON by default. It is not a big deal, but still might result in unwanted behavior if someone builds SRT core as an external in 3rd party project e.g. without using our cmake. Not sure if someone does that though.

Reply:
@maxlovic,
it would be great and I'm sure it's possible to add busy_wait as well as ENABLE_UNEVEN_PACING as srt parameter(like latency, maxbw etc) for given connection and don't move them as compilation params.
When you use tools like ffmpeg or srt-trasmit you can have two versions without any problems but if you add SRT into the media server like we do you have a problem. Some people will definitely want to have some streams with default behavior and some streams with busy_wait.
Moreover once you introduce new build options you will have 2^n possible builds where n is a number of features and you will have to build srt with particular combination many times.
Options instead of compilation params is the only option if we want to have single SRT build with all possible use cases in it.

Reply:
@alexpokotilo Good point. Adding busy waiting as an option might introduce performance burden.
This PR is first of all to introduce a build option for users in 1.3.3.
Then PR #678 is scheduled to go in 1.3.4.
Then there are several ways to go. One way is that you suggested. One more is suggested in #673, where busy waiting is used for small wait intervals together with normal sleep.
I am also thinking about one timer shared between several sockets.